src/App.tsx

```diff
@@ -2,14 +2,51 @@ import { BrowserRouter as Router, Routes, Route } from "react-router";
 import Home from "./pages/Home";
 import Favorites from "./pages/Favorites";
 import { PageLayout } from "./layouts/PageLayout";
+import { useState, useEffect } from "react";
 
 function App() {
+  const [isMobileView, setIsMobileView] = useState(false);
+  const [selectedCharacter, setSelectedCharacter] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobileView(window.innerWidth < 640);
```

Comment:
Try to avoid magic numbers:

```suggestion
      setIsMobileView(window.innerWidth < MOBILE_WIDTH);
```

ref: https://refactoring.guru/replace-magic-number-with-symbolic-constant